source/link/linker.cpp

```diff
@@ -696,6 +724,44 @@ spv_result_t VerifyLimits(const MessageConsumer& consumer,
   return SPV_SUCCESS;
 }
 
+spv_result_t FixFunctionCallTypes(opt::IRContext& context,
+                                  const LinkageTable& linkings) {
+  auto mod = context.module();
+  const auto type_manager = context.get_type_mgr();
+
+  for (auto& func : *mod) {
+    func.ForEachInst([&](Instruction* inst) {
+      if (inst->opcode() != spv::Op::OpFunctionCall) return;
+      opt::Operand& target = inst->GetInOperand(0);
+
+      // only fix calls to imported functions
+      auto linking = std::find_if(
+          linkings.begin(), linkings.end(), [&](const auto& entry) {
+            return entry.exported_symbol.id == target.AsId();
+          });
+      if (linking == linkings.end()) return;
+
+      Function* exported_symbol_type =
+          type_manager->GetType(linking->exported_symbol.type_id)->AsFunction();
+
+      for (uint32_t i = 1; i < inst->NumInOperands(); ++i) {
+        const Type* arg_type = exported_symbol_type->param_types()[i - 1];
+        if (arg_type->kind() != Type::kPointer) continue;
+
+        const auto ptr_type = arg_type->AsPointer();
+        opt::Operand& arg = inst->GetInOperand(i);
+        // TODO:
+        //  - get target type id
+        //  - insert instruction
+        //  - replace operand
+        new Instruction(&context, spv::Op::OpBitcast, ..., 0u, {{SPV_OPERAND_TYPE_ID, arg.AsId()}});
```

Comment:
I do not think this will compile as-is, and a `res_id` of 0 seems wrong.

Reply:
yeah, that's why I put the comments on top because I was kinda stuck there